src/student_success_tool/modeling/feature_selection.py

```diff
@@ -151,49 +179,98 @@ def drop_collinear_features_iteratively(
         mlflow.autolog(disable=True)
 
     Args:
-        features: dataframe of features to assess for multicollinearity
+        df
+        threshold: Variance Inflaction Factor (VIF) above which columns are considered
+            "multicollinear" and dropped from ``df`` . A VIF of 1 indicates no correlation
+            with any other feature; a VIF of 10 is considered "very high" and contributing
+            to multicollinearity.
         force_include_cols
 
     Returns:
-        features not considered collinear according to a VIF threshold of 10
+        features not considered collinear according to a VIF threshold
     """
-
     np.seterr(divide="ignore", invalid="ignore")
+    force_include_cols = force_include_cols or []
+
+    df_features = df.select_dtypes(include="number")
+    if df_features.shape[1] == 0:
+        LOGGER.warning("no numeric columns found, so no collinear features to drop")
+        return df
 
-    numeric_features = features.select_dtypes(include="number")
-    imp = SimpleImputer(missing_values=np.nan, strategy="mean")
-    numeric_features_imputed = pd.DataFrame(imp.fit_transform(numeric_features))
-    numeric_features_imputed.columns = numeric_features.columns.values
```

Comment:
Oops, I forgot to answer this question. It was needed because `SimpleImputer.fit_transform()` was returning a numpy array instead of a dataframe, so you needed to re-add the column names back in. I changed that in the line below this comment by leveraging `.set_output(transform="pandas")` -- now the imputer returns a dataframe with the columns already populated as you'd expect. This is a fairly new sklearn feature, and it's lovely! https://scikit-learn.org/stable/auto_examples/miscellaneous/plot_set_output.html#sphx-glr-auto-examples-miscellaneous-plot-set-output-py